config/src/main/java/org/springframework/security/config/annotation/web/configurers/oauth2/server/resource/OAuth2ResourceServerConfigurer.java

```diff
@@ -209,10 +221,27 @@ public JwtConfigurer jwkSetUri(String uri) {
 			return this;
 		}
 
+		public JwtConfigurer jwtConverter(Converter<Jwt, ? extends AbstractAuthenticationToken> jwtConverter) {
```

Comment:
This name does not align with the current `decoder(JwtDecoder decoder)` method.

However, I'm not sure if `converter(Converter<Jwt, ? extends AbstractAuthenticationToken> jwtConverter)` makes sense either?

I'm wondering if `authenticationConverter(Converter<Jwt, ? extends AbstractAuthenticationToken> converter)` works?

Reply:
Yeah, let's circle back with @rwinch on this. He mentioned that `jwtDecoder` would be more correct than `decoder`. I'm not sure I've captured his reasoning yet, but, yes, if we are going to stick with `decoder`, then this should be `converter` (or `authenticationConverter` as you suggested).

Btw, it appears that we are introducing the notion of `AuthenticationConverter` in another PR, which converts from a `ServerWebExchange` to a `Mono<Authentication>`. Just something to keep in mind as we consider the name here.

Reply:
@rwinch would `ServerAuthenticationConverter`'s contract imply that the hypothetical `AuthenticationConverter` would be `HttpServletRequest -> Authentication`, then?

Reply:
I was mostly clarifying the name. I don't think it really doesn't matter too much from this conversations perspective.

> I think that JwtConverter or JwtAuthenticationConverter would be acceptable with a slight preference to JwtAuthenticationConverter.

I agree. It is converting from a Jwt to an authentication token so I find JwtAuthenticationConverter useful.

Reply:
For historical purposes, I'll mention that I made a comment earlier about not wanting to go with `AuthenticationConverter` for `Jwt` -> `AbstractAuthenticationToken` conversions since `ServerAuthetnicationConverter` is semantically deviant from that. Then, I posited that `JwtAuthenticationConverter` might be a better name.

*However* I had misread the conversation, and once I realized it didn't apply, I deleted my comment, but not before Rob speedily replied. :)

Sorry for any confusion for those trying to figure out who Rob was quoting.